internal/maintenance/queue_maintainer.go

```diff
@@ -99,7 +99,12 @@ func (m *QueueMaintainer) Start(ctx context.Context) error {
 // reflection and potential for panics.
 func GetService[T startstop.Service](maintainer *QueueMaintainer) T {
 	var kindPtr T
-	return maintainer.servicesByName[reflect.TypeOf(kindPtr).Elem().Name()].(T) //nolint:forcetypeassert
+	return maintainer.servicesByName[serviceName(kindPtr)].(T) //nolint:forcetypeassert
+}
+
+func serviceName(service startstop.Service) string {
+	elem := reflect.TypeOf(service).Elem()
+	return elem.PkgPath() + "." + elem.Name()
```

Comment:
I made this change to the make the internal names more unique because I had originally had two pro maintenance processes with the same name in different packages, and I lost a fair bit of time trying to figure out why one of them wasn't working. I don't think there are downsides/risks here other than just the internal name being the longer & more complicated path+type combo since it's only used internally.